Add section "Boot loader management and file structure" (#4422)
theforeman/foreman-documentation: guides/common/modules/ref_bootloader-types.adoc

:_mod-docs-content-type: REFERENCE

[id="ref_bootloader-types_{context}"]
= Boot loader types in {Project}

> This section describes the PXE boot loaders supported by {Project} and explains how each type is used during network provisioning, depending on firmware type, protocol, and boot environment.

Reviewer: ```suggestion
{Project} supports multiple PXE boot loaders. You can select a PXE boot loader during network provisioning, depending on your firmware type, protocol, and boot environment.
```

. PXE Loader Types and Boot Protocols

[cols="1,1,1,1", options="header"]
|===
| PXE Loader (UI Label)
| Category
| Firmware
| Protocol

| None
| None
| -
| -

| PXELinux BIOS
| PXELinux
| BIOS
| DHCP → TFTP

| PXELinux UEFI
| PXELinux
| UEFI
| DHCP → TFTP

| Grub UEFI
| PXEGrub (Legacy)
| UEFI
| DHCP → TFTP

| Grub2 BIOS
| PXEGrub2 (TFTP)
| BIOS
| DHCP → TFTP

| Grub2 ELF
| PXEGrub2 (TFTP)
| BIOS
| DHCP → TFTP

| Grub2 UEFI
| PXEGrub2 (TFTP)
| UEFI
| DHCP → TFTP

| Grub2 UEFI SecureBoot
| PXEGrub2 (TFTP)
| UEFI
| DHCP → TFTP

| Grub2 UEFI HTTP
| Grub2 (HTTP)
| UEFI
| DHCP → HTTP

| Grub2 UEFI HTTPS
| Grub2 (HTTPS)
| UEFI
| DHCP → HTTPS

| Grub2 UEFI HTTPS SecureBoot
| Grub2 (HTTPS)
| UEFI
| DHCP → HTTPS

| iPXE Embedded
| iPXE Direct (HTTP)
| BIOS / UEFI
| DHCP → HTTP

| iPXE UEFI HTTP
| iPXE Direct (HTTP)
| UEFI
| DHCP → HTTP

| iPXE Chain BIOS
| iPXE Chainload
| BIOS
| DHCP → TFTP → HTTP

| iPXE Chain UEFI
| iPXE Chainload
| UEFI
| DHCP → TFTP → HTTP
|===

This table lists the available PXE Loader categories and describes how each type delivers boot loaders and installation files during network provisioning.

> . PXE Loader categories

Reviewer: ```suggestion
.PXE Loader categories
```

[cols="30%,70%",options="header"]
|===
| Category | Description

| PXELinux
| Syslinux-based loaders for BIOS and UEFI environments.

| PXEGrub (Legacy)
| Early GRUB-based loader for UEFI systems.

| PXEGrub2 (TFTP)
| GRUB2-based loader supporting both BIOS and UEFI environments (including Secure Boot via `shim`).

| Grub2 (HTTP/HTTPS)
| GRUB2-based UEFI boot loader fetched over HTTP or HTTPS using the {SmartProxy}’s HTTPBoot service. 

| iPXE Chainload
| A two-stage boot process in which the client first downloads an iPXE binary (`undionly-ipxe.0` or `ipxe.efi`) via TFTP.
Once loaded, iPXE retrieves {Project}’s iPXE script and installation files over HTTP, enabling flexible configuration and network boot via HTTP.

| iPXE Direct (HTTP)
| Boots directly via embedded or firmware-level iPXE using only HTTP.

| None
| {Project} does not specify a boot loader.

> |===

Author: Thank you for the suggestion.  
I agree that a description list is a better fit for this section than a table.  
I've applied your proposed change and updated the structure accordingly.

> 

Reviewer 2: I would not strip down categories to pxelinux, GRUB, GRUB2, iPXE
